include/yaramod/utils/utils.h

```diff
@@ -48,12 +49,36 @@ template <typename T>
 std::string numToStr(const T num, std::ios_base &(*format)(std::ios_base&) = std::dec, bool showbase = false, bool toUpper = false)
 {
 	std::ostringstream os;
+	if (std::is_floating_point<T>::value)
+		os.precision(std::numeric_limits<T>::digits10 - 1);
+
+	os << std::fixed;
 	if (toUpper)
 		os << std::uppercase;
 	if (showbase)
 		os << format << std::showbase << num;
 	else
 		os << format << num;
+
+	if (std::is_floating_point<T>::value)
+	{
+		std::string value = os.str();
+		auto comma = value.find('.');
+		if (comma == std::string::npos)
+		{
+			return value.append(".0");
+		}
+		if (comma == value.length() - 1)
+		{
+			return value.append("0");
+		}
+		if (comma == value.length() - 2)
+		{
+			return value;
+		}
+		auto zero = value.find_last_not_of('0');
```

Comment:
Could you please post some comments here of what you are trying to achieve because it's not obvious from the get go.

Reply:
There are few cases:
* `comma == std::string::npos`: no comma in string so append '.0`
* `comma == value.length() - 1`: Comma is last character so append 0
* `comma == value.length() - 2`: Value have one decimal place, so don't change value
* `else`: trim zeros from end of string presentation of double.  So instead of 10.00000000000000 function will return 10.0

Reply:
Okay and why can't we have it in the code also to state the intent? Please put it there. Thanks.

Reply:
Added comments.